src/tree.h

```diff
@@ -17,15 +17,15 @@
 struct git_tree_entry {
 	uint16_t attr;
 	uint16_t filename_len;
-	git_oid oid;
-	bool pooled;
-	char filename[GIT_FLEX_ARRAY];
+	const git_oid *oid;
+	const char *filename;
 };
 
 struct git_tree {
 	git_object object;
+	git_odb_object *odb_obj;
+	git_array_t(git_tree_entry) entries_arr;
 	git_vector entries;
```

Comment:
Cool.  We had talked once about how we do a sort on tree entries instead of trusting that the entries in the tree were written correctly, and how we should ⚡ this behavior.

I'm happy to merge this as-is but I think we should drop the vector when time permits.  (I would also be happy to tackle this later.)